packages/box/src/handlers/ipfs-upload.ts

```diff
@@ -0,0 +1,141 @@
+import { Request, Response } from "express";
+import multer from "multer";
+import fs from "fs";
+import FormData from "form-data";
+import axios from "axios";
+import {
+    ipfsUploadRequest,
+    IpfsUploadRequest,
+    BadRequestError,
+    handleErrors,
+    parseActionIpfsCid,
+    validateActionFile,
+} from "@edenos/common";
+
+import { ipfsConfig, validUploadActions } from "../config";
+import { eosJsonRpc, eosDefaultApi } from "../eos";
+
+export const ipfsUploadConfigHandler = multer({ dest: "tmp/" }).single("file");
+
+export const ipfsUploadHandler = async (req: Request, res: Response) => {
+    const { file, body } = req;
+    try {
+        if (!file) {
+            throw new BadRequestError(["no file to upload"]);
+        }
+
+        if (!body.data) {
+            throw new BadRequestError(["missing transaction data"]);
+        }
+
+        const parsedRequest = ipfsUploadRequest.safeParse(
+            JSON.parse(body.data)
+        );
+        if (parsedRequest.success !== true) {
+            throw new BadRequestError(parsedRequest.error.flatten());
+        }
+
+        const requestData: IpfsUploadRequest = parsedRequest.data;
+
+        if (!requestData.syncUpload) {
+            // We need to implement a background job mechanism to make sure the
+            // file can be upload asynchronously without blocking the client.
```

Comment:
We should probably mark this obviously with a TODO so we can easily find this and be reminded to come back to it

Reply:
done! 👍